scipy/sparse/linalg/_special_sparse_arrays.py

```diff
@@ -511,3 +513,391 @@ def _adjoint(self):
+        return 1. / aranp1
+
+    def toarray(self):
+        n = self.shape[0]
+        aranp1 = np.arange(1, n + 1, dtype=self.dtype)
+        aranp1_inv = 1. / aranp1
+        return np.diag(aranp1_inv)
+
+    def _matvec(self, x):
+        n = self.shape[0]
+        aranp1_inv = 1. / np.arange(1, n + 1)
+        # linearoperator requires 2D array
+        if len(x.shape) == 1:
+            x = x.reshape(-1, 1)
+        return aranp1_inv[:, np.newaxis] * x
+
+    def _matmat(self, x):
+        return self._matvec(x)
+
+    def _adjoint(self):
+        return self
+
+    def _transpose(self):
+        return self
+
+
+class MikotaK(LinearOperator):
+    """
+    Construct a stiffness matrix in various formats of Mikota pair.
+
+    The stiffness matrix `K` is square real tri-diagonal symmetric
+    positive definite with integer entries. 
+
+    Parameters
+    ----------
+    shape : tuple of int
+        The shape of the matrix.
+    dtype : dtype
+        Numerical type of the array. Default is ``np.int32``.
+
+    Methods
+    -------
+    toarray()
+        Construct a dense array from Laplacian data
+    tosparse()
+        Construct a sparse array from Laplacian data
+    tobanded()
+        The format for banded symmetric matrices,
+        i.e., (2, n) ndarray with 2 upper diagonals
+        placing the main diagonal at the bottom.
+    """
+    def __init__(self, shape, dtype=np.int32):
+        self.shape = shape
+        self.dtype = dtype
+        super().__init__(dtype, shape)
+
+    def tosparse(self):
+        from scipy.sparse import diags
+        n = self.shape[0]
+        y = - np.arange(n - 1, 0, -1, dtype=self.dtype)
+        z = np.arange(2 * n - 1, 0, -2, dtype=self.dtype)
+        return diags([y, z, y], [-1, 0, 1], shape=(n, n))
+
+    def tobanded(self):
+        n = self.shape[0]
+        y = - np.arange(n - 1, 0, -1, dtype=self.dtype)
+        z = np.arange(2 * n - 1, 0, -2, dtype=self.dtype)
+        return np.array([np.pad(y, (1, 0), 'constant'), z])
+
+    def toarray(self):
+        return self.tosparse().toarray()
+
+    def _matvec(self, x):
+        n = self.shape[0]
+        x = x.reshape(n, -1)
+        kx = np.zeros_like(x)
+        y = - np.arange(n - 1, 0, -1)
+        z = np.arange(2 * n - 1, 0, -2)
+        kx[0, :] = z[0] * x[0, :] + y[0] * x[1, :]
+        kx[-1, :] = y[-1] * x[-2, :] + z[-1] * x[-1, :]
+        kx[1: -1, :] = (y[:-1, None] * x[: -2,:]
+                        + z[1: -1, None] * x[1: -1, :]
+                        + y[1:, None] * x[2:, :])
+        return kx
+
+    def _matmat(self, x):
+        return self._matvec(x)
+
+    def _adjoint(self):
+        return self
+
+    def _transpose(self):
+        return self
+
+
+class Mikota_pair:
+    """
+    Construct the Mikota pair of matrices in various formats and
+    eigenvalues of the generalized eigenproblem with them.
+
+    The Mikota pair of matrices [1, 2]_ models a vibration problem
```

Comment:
The docstings here are for reference only - the actual documentation is never generated since there is no public API. The doctrings are basically just comments in this PR. 

I do not know if it works, i.e., we can just keep it as is.